g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/date/DateFormatService.java

```diff
@@ -39,4 +64,54 @@ public String formatDate(String locale, long date, String pattern) throws L2APIE
 		}
 	}
 
+	
+	/**
+   	 * @param locale
+   	 * @param default territory
+   	 * @return matching locale TimeZoneName
+   	 */
+	@Override
+	public TimeZoneName getTimeZoneName(String oldLocale, boolean defaultTerritory) throws L2APIException {
+		// TODO Auto-generated method stub
+		String locale = oldLocale.replace("_", "-");
```

Comment:
I don't see where to use the oldLocale, is it a invalid parameter?

Reply:
oldLocale.replace("_", "-"),  this variable use to record the locale string that Illegal request local string parameter and reply to
client side

Reply:
done